tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/BlockingMiddleware.cs

```diff
@@ -116,18 +116,19 @@ internal async Task Invoke(HttpContext context)
             {
                 await _next(context).ConfigureAwait(false);
             }
-            catch (BlockException e)
+            catch (Exception e) when (e is BlockException || e.InnerException is BlockException)
```

Comment:
You might want to consider using a recursive approach to check the nested InnerExceptions, similar to what we do [here](https://github.com/DataDog/dd-trace-dotnet/blob/68e01ee9bce3ad24f94d875f2dc5ad21248aaa03/tracer/src/Datadog.Trace/Util/Http/HttpExceptionExtensions.cs) for `SocketException`.

For example, you could have a helper like this:

```csharp
private static BlockException? GetBlockException(Exception exception)
{
    while (exception is not null)
    {
        if (exception is BlockException b)
        {
            return b;
        }

        exception = exception.InnerException;
    }

    return null;
}
```

Then you could use it in the catch block like this:
```suggestion
            catch (Exception e) when (GetBlockException(e) is { } blockException)
```

(and remove the `var blockException...` from the method block)

Reply:
thanks for letting me know, fixed in https://github.com/DataDog/dd-trace-dotnet/pull/3944/commits/cc6a1c2774cee57618d22d829b224b9d31880b99

Actually, I came across the function `GetBaseException`, do you think it [does exactly that](https://learn.microsoft.com/en-us/dotnet/api/system.exception.getbaseexception?view=net-7.0)?
Just tried for example:
```var grandChild = new ArgumentNullException();
var child = new ArgumentException("child", grandChild);
var parent = new Exception("parent", child);
var result = parent.GetBaseException();
Console.WriteLine(result);
```

and result is indeed the `grandChild `..

Reply:
I went ahead with `GetBaseException` in the end, .NET source code [seems to be doing what we want](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/mscorlib/system/exception.cs#L249) in fact

Reply:
That will work as long as we never set the `InnerException` on the `BlockException`, which may be the case now, but won't _necessarily_ always be the case. Personally I would go with my suggestion to be safe, but I'm biased 😄 Alternatively, add a comment to `BlockException` to say why to never set `InnerException` 🙂